transcode/transcode.go

```diff
@@ -165,6 +155,68 @@ func RunTranscodeProcess(transcodeRequest TranscodeSegmentRequest, streamName st
 		return outputs, segmentsCount, err
 	}
 
+	// Transmux received segments from T into a single mp4
+	if transcodeRequest.GenerateMP4 {
+		for rendition, segments := range renditionList.RenditionSegmentTable {
+			// a. create folder to hold transmux-ed files in local storage temporarily
+			err := os.MkdirAll(TRANSMUX_STORAGE_DIR, 0700)
+			if err != nil && !os.IsExist(err) {
+				log.Log(transcodeRequest.RequestID, "failed to create temp dir for transmuxing", "dir", TRANSMUX_STORAGE_DIR, "err", err)
+				return outputs, segmentsCount, err
+			}
+
+			// b. create a single .ts file for a given rendition by concatenating all segments in order
+			concatTsFileName := filepath.Join(TRANSMUX_STORAGE_DIR, transcodeRequest.RequestID+"_"+rendition+".ts")
+			totalBytes, err := video.ConcatTS(concatTsFileName, segments)
+			if err != nil {
+				log.Log(transcodeRequest.RequestID, "error concatenating .ts", "file", concatTsFileName, "err", err)
+				continue
+			}
+			defer os.Remove(concatTsFileName)
```

Comment:
we just need to move this line onto the line above the call to `video.ConcatTS` so that if there's an error from that function on any of the individual writes then we will still clean up the file. It doesn't matter if we try to delete it before it gets created as we're not doing anything with the error from `os.Remove` anyway.

Reply:
Good catch - fixed.